app/models/search.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Search
+  include ActiveModel::Model
+
+  attr_accessor :query, :planning_application_ids
+
+  validates :query, presence: true
+
+  def results
+    if valid?
+      planning_applications.where("reference LIKE ?", "%#{query}%")
```

Comment:
🎉

Reply:
I think we'll actually want to add a LOWER function here too so we can be case insensitive

Reply:
which means we may need to add that to the index on reference i.e. `add_index :planning_applications, "lower(reference)"`

Reply:
oo...good thinking!